util/class.tx_mksanitizedparameters_util_RegularExpression.php

```diff
@@ -30,8 +30,6 @@ class tx_mksanitizedparameters_util_RegularExpression
     /**
      * @param string $pattern
      * @param mixed  $value
-     *
-     * @return string
```

Comment:
Wouldn't be better to specify the correct return type, rather to delete them?

Reply:
You're right again. This method is called in **tx_mksanitizedparameters_sanitizer_Alnum** and **tx_mksanitizedparameters_sanitizer_Alpha**. Those methods only called in TestCases.

Can I change the return type of those methodes to null|string|string[] ?

But I think, this is also the wrong way, because those methods cast $value to string. Only one string can be returned like this.

Reply:
I think null|string|array would be correct as that's what preg_replace returns as stated in the PHP documentation (https://php.net/manual/de/function.preg-replace.php).

Reply:
@hbochmann 
You're right. I will do that!

Reply:
@hbochmann 

But please bear in mind, that if change the typhint of **callPregReplace**, I also must change the typhints of those methods, which call this.

Those methods deliver a value, which is parsed to string. So those methods only can return a string (see the documentation).

There I can use a annotation, but parsing is not possible.